Makefile

```diff
@@ -1,15 +1,22 @@
 .PHONY: build-rust-optimizer build-workspace-optimizer build publish-rust-optimizer publish-workspace-optimizer publish
 
+DOCKER_NAME_BASE_OPTIMIZER := "cosmwasm/base-optimizer"
 DOCKER_NAME_RUST_OPTIMIZER := "cosmwasm/rust-optimizer"
 DOCKER_NAME_WORKSPACE_OPTIMIZER := "cosmwasm/workspace-optimizer"
-DOCKER_TAG := 0.11.0
+DOCKER_TAG := 0.12.0
 
-build-rust-optimizer:
+build-base-optimizer:
+	docker build -t $(DOCKER_NAME_BASE_OPTIMIZER):$(DOCKER_TAG) --file base-optimizer.Dockerfile .
+
+build-rust-optimizer: build-base-optimizer
 	docker build -t $(DOCKER_NAME_RUST_OPTIMIZER):$(DOCKER_TAG) --file rust-optimizer.Dockerfile .
 
-build-workspace-optimizer:
+build-workspace-optimizer: build-base-optimizer
 	docker build -t $(DOCKER_NAME_WORKSPACE_OPTIMIZER):$(DOCKER_TAG) --file workspace-optimizer.Dockerfile .
 
+publish-base-optimizer: build-base-optimizer
+	docker push $(DOCKER_NAME_BASE_OPTIMIZER):$(DOCKER_TAG)
```

Comment:
Is it necessary to publish this? Isn't there and solution fo local deduplication?

Reply:
It's not necessary to publish it at all. I'll remove it.

Reply:
Publishing makes sense if we decouple building `base-optimizer` from the others. Given that here building is coupled, publishing is not necessary.